pinot-server/src/main/java/org/apache/pinot/server/api/resources/TablesResource.java

```diff
@@ -1033,7 +1033,10 @@ public TableSegmentValidationInfo validateTableSegmentState(
             case SegmentStateModel.CONSUMING:
               // Only validate presence of segment
               if (segmentDataManager == null) {
-                return new TableSegmentValidationInfo(false, -1);
+                String invalidReason = String.format(
+                    "Segment %s is in CONSUMING state, but segmentDataManager is null", segmentName);
+                LOGGER.error(invalidReason);
```

Comment:
Let's not log error here. This is normal case, and returning the invalid reason through response should be good enough. Same for other places